code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -164,9 +164,10 @@
 	if(!user.client)
 		return
 	if(user.s_active != src) // Switching from another container
-		for(var/obj/item/I in src)
-			if(I.on_found(user)) // For mouse traps and such
-				return // If something triggered, don't open the UI
+		if(!isobserver(user)) //Don't want ghosts setting off the fun stuff
```

Comment:
This check should really be put into `/obj/item/assembly/mousetrap/on_found`. Return from the proc if they're an observer.

Reply:
I did not put it there, in the event other on_founds are added, and due to facehuggers also using the same proc.

Reply:
Ugh alright, I don't like how this needs to be snowflaked, but I don't have a better idea right now.